packages/frontend/src/components/EditorSettings/FlowTransfer/DisallowRequestInfobox.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback } from 'react'
+import { useMutation } from '@apollo/client'
+import { Flex, Text } from '@chakra-ui/react'
+import { Button, Infobox, useToast } from '@opengovsg/design-system-react'
+import { UPDATE_FLOW_TRANSFER_STATUS } from 'graphql/mutations/update-flow-transfer-status'
+import { GET_PENDING_FLOW_TRANSFER } from 'graphql/queries/get-pending-flow-transfer'
+
+interface FlowTransferWarningProps {
+  flowTransferId: string
+  requestedEmail: string
+}
+
+export default function FlowTransferWarning(props: FlowTransferWarningProps) {
+  const { flowTransferId, requestedEmail } = props
+  const toast = useToast()
+  const [updateFlowTransferStatus] = useMutation(UPDATE_FLOW_TRANSFER_STATUS)
+
+  const onFlowTransferStatusUpdate = useCallback(async () => {
+    await updateFlowTransferStatus({
+      variables: {
+        input: {
+          id: flowTransferId,
+          status: 'cancelled',
+        },
+      },
+      refetchQueries: [GET_PENDING_FLOW_TRANSFER],
```

Comment:
hmmm... ok I think i see why you did a separate query.

I personally think code complexity due to 2 queries is not worth any network load savings we get from being able to refetch less data due to a separate query. 

Let me know what you think.